src/Nethermind/Nethermind.Sockets/CounterStream.cs

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.IO;
+
+namespace Nethermind.Sockets;
+
+public class CounterStream : Stream
+{
+    private readonly Stream _stream;
+
+    public CounterStream(Stream stream)
```

Comment:
NP: No library for this?

Reply:
Probably not/not worth it we already have `CountingTextReader`, `CountingTextWriter`

Reply:
> NP: No library for this?
Searching online I found several SO questions showing how to implement such feature, but no standalone library.